src/sys/windows.rs

```diff
@@ -938,9 +938,11 @@ pub(crate) fn unix_sockaddr(path: &Path) -> io::Result<SockAddr> {
         }
 
         storage.sun_family = crate::sys::AF_UNIX as sa_family_t;
+        // SAFETY: casting `[u8]` to `[i8]` is safe.
+        let b = unsafe { &*(bytes as *const [u8] as *const [i8]) };
```

Comment:
Seems a bit fragile to rely on char being i8.

Reply:
Oh, the type itself actually uses `i8`, not a type alias: https://docs.rs/windows-sys/latest/windows_sys/Win32/Networking/WinSock/struct.SOCKADDR_UN.html.